bin/create-digests.py

```diff
@@ -117,11 +117,175 @@ def collect_digests(digests, subdir, exclude=[]):
+            digest_totals[election_name][f'_{total_key}_from_tickets'] = round(election_total_from_tickets,2)
+
 def main():
     digests = {}
     build_dir = 'build'
     filepath = f'{build_dir}/digests.json'
     collect_digests(digests, build_dir, exclude=[filepath])
+
+    # contribution totals
+    add_totals(digests, total_key='total_contributions')
+    add_tickets_total(digests, ticket_type='candidates', total_key='total_contributions')
+    add_tickets_total(digests, ticket_type='referendum_opposing', total_key='total_contributions')
+    add_tickets_total(digests, ticket_type='referendum_supporting', total_key='total_contributions')
+    add_combined_tickets_total(digests, total_keys=['total_contributions'])
+    add_elections_total(digests, total_key='total_contributions')
+
+    # contributions by type totals
+    add_totals(digests, total_key='contributions_by_type', total_group_key='total_contributions')
+    add_tickets_total(digests, ticket_type='candidates', total_key='contributions_by_type', total_group_key='total_contributions')
+    add_tickets_total(digests, ticket_type='referendum_opposing', total_key='contributions_by_type', total_group_key='total_contributions')
+    add_tickets_total(digests, ticket_type='referendum_supporting', total_key='contributions_by_type', total_group_key='total_contributions')
+    add_combined_tickets_total(digests, total_keys=['contributions_by_type'], total_group_key='total_contributions')
+    add_elections_total(digests, total_key='contributions_by_type', total_group_key='total_contributions')
+
+    # small contributions totals
+    add_tickets_total(digests, ticket_type='candidates', total_key='total_small_contributions', total_group_key='total_contributions')
+    
+    # contributions by source totals
+    add_totals(digests, total_key='total_contributions_by_source', total_group_key='total_contributions')
+    add_tickets_total(digests, ticket_type='candidates', total_key='contributions_by_origin', total_group_key='total_contributions')
+    add_tickets_total(digests, ticket_type='referendum_opposing', total_key='contributions_by_region', total_group_key='total_contributions')
+    add_tickets_total(digests, ticket_type='referendum_supporting', total_key='contributions_by_region', total_group_key='total_contributions')
+    add_combined_tickets_total(digests, total_keys=['total_contributions_by_source','contributions_by_origin','contributions_by_region'], total_group_key='total_contributions')
+    add_elections_total(digests, total_key='total_contributions_by_source', total_group_key='total_contributions')
+
+    # expenditure totals
+    add_tickets_total(digests, ticket_type='candidates', total_key='total_expenditures')
+
+    # expenditures by type totals
+    add_tickets_total(digests, ticket_type='candidates', total_key='expenditures_by_type', total_group_key='total_expenditures')
+
+    # some totals that seems to have nothing
```

Comment:
What do you mean "have nothing"

Reply:
The json files show empty values (like braces with nothing in them)